var/spack/repos/builtin/packages/fzf/package.py

```diff
@@ -27,22 +30,32 @@ class Fzf(MakefilePackage):
     version('0.16.9',   sha256='dd9434576c68313481613a5bd52dbf623eee37a5c87f7bb66ca71ac8add5ff94')
     version('0.16.8',   sha256='daef99f67cff3dad261dbcf2aef995bb78b360bcc7098d7230cb11674e1ee1d4')
 
-    depends_on('go@1.11:')
+    depends_on('go@1.11:', type='build')
 
     variant('vim', default=False, description='Install vim plugins for fzf')
 
-    patch("github_mirrors.patch")
+    patch("github_mirrors.patch", when='@:0.17.5')
+
+    @classmethod
+    def determine_version(cls, exe):
+        return Executable(exe)('--version', output=str, error=str).rstrip()
 
     def build(self, spec, prefix):
-        glide_home = os.path.join(self.build_directory, 'glide_home')
-        os.environ['GLIDE_HOME'] = glide_home
-        shutil.rmtree(glide_home, ignore_errors=True)
-        os.mkdir(glide_home)
-        super(Fzf, self).build(spec, prefix)
+        if self.version <= Version('0.17.5'):
+            glide_home = os.path.join(self.build_directory, 'glide_home')
+            os.environ['GLIDE_HOME'] = glide_home
+            shutil.rmtree(glide_home, ignore_errors=True)
+            os.mkdir(glide_home)
+            super(Fzf, self).build(spec, prefix)
+        else:
+            make()
 
     def install(self, spec, prefix):
```

Comment:
I would simplify this like so:
```python
@run_after('install')
def post_install(self):
    mkdir(self.prefix.bin)
    install('bin/fzf', self.prefix.bin)
...
```

Reply:
The behaviour of `make install` for fzf is just to move the binary into a `bin/`  (`install: bin/fzf`) directory so I run into the follow:

```shell
==> fzf: Executing phase: 'install'
==> Error: InstallError: Install failed for fzf.  Nothing was installed!

/home/paul/Development/60_Personal/spack/lib/spack/spack/package.py:1616, in sanity_check_prefix:
       1613        installed = set(os.listdir(self.prefix))
       1614        installed.difference_update(
       1615            spack.store.layout.hidden_file_paths)
  >>   1616        if not installed:
       1617            raise InstallError(
       1618                "Install failed for %s.  Nothing was installed!" % self.name)

See build log for details:
  /tmp/paul/spack-stage/spack-stage-fzf-0.22.0-uxagnwuta5dcwujn5ak7hwc33pgknooo/spack-build-out.txt
```

I simplified it a little and tested the build against older versions with success. Do you think those changes are sufficient?

Reply:
If you move the `mkdir` and `install` to the `post_install` function, you can get rid of the `install` phase entirely and rely on the default, which is to run `make('install')`.

Reply:
Apologies, I'm not a big user of fzf just wanted to update the package while adding the `determine_version` functionality so I could be wrong on some aspects (so to be safe I looked over the project again).

When I make the change you recommend I still run into the error above. In the case of fzf's makefile the build (`make`) and install (`make install`) all occur in the root directory (`spack-stage/.../spack-src`) with seemingly no way to influence these. This is slightly accounted for by the project's install script; however, that makes a number of changes to the user's profile which seems unwise.

If I'm missing something with the package please let me know as I'm unsure of any avenue I can use to align with your recommendation.

Reply:
Let's just leave it as is for now, thanks for looking into it.